Enable use of EventeListeners to inform when link is enqueued and dequeued

zesarone opened the pull request:
Used this implementation in our app and it helped us along with notifying the UI to act accordingly.

Can be used as follows
```
QueueLink.addLinkQueueEventListener("feedback", "enqueue", (item: any) => {
// react to event as is appropriate to your application at this event. 
    });
```

Please merge/edit/ignore if you find this useful and aligning with project plan.

emilhammarstedt committed:
Add Eventlistener feature
- It is now possible to add an eventlistener to be notified after beeing queued and before beeing dequeued for further processing.

emilhammarstedt committed:
Code clean up

emilhammarstedt committed:
Code clean up

helfer commented:
Sorry for the delay @zesarone! I think this makes sense and I'd be open to merging it, but can you tell me a bit more about your use case first? Are you using it to build offline persistence, or do something else?

fishyFrogFace committed:
Removal of event listeners (#2)

* Removal of old listeners

- Return a GUID on creation of a listener
- Function for removing the listener by passing GUID, opName and event

* Remove tslint rules that are no longer in use

zesarone commented:
@helfer Our use case is as you mention to do offline persistence. the idea is to have a convenient access point to trigger actions in the UI and others when things get queued. Got some other changes from @fishyFrogFace in to handle cleanup of listeners etc.

zesarone committed:
Merge branch 'helfer:master' into master
